runtime/src/main/java/org/corfudb/runtime/collections/StreamManager.java

```diff
@@ -0,0 +1,225 @@
+package org.corfudb.runtime.collections;
+
+import com.google.protobuf.Message;
+
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.stream.Collectors;
+
+import javax.annotation.Nonnull;
+
+import org.corfudb.protocols.logprotocol.MultiObjectSMREntry;
+import org.corfudb.protocols.wireprotocol.ILogData;
+import org.corfudb.runtime.CorfuRuntime;
+import org.corfudb.runtime.exceptions.StreamSubscriptionException;
+import org.corfudb.runtime.view.Address;
+import org.corfudb.runtime.view.ObjectsView;
+import org.corfudb.runtime.view.StreamOptions;
+import org.corfudb.runtime.view.TableRegistry;
+import org.corfudb.runtime.view.stream.IStreamView;
+
+/**
+ * A simple thread based subscription engine where each subscriber or listener gets
+ * a thread which will listen on the tables of interest to it and moves at a rate
+ * which is the same as the consumption.
+ *
+ * Created by hisundar on 04/28/2020.
+ */
+@Slf4j
+public class StreamManager {
+
+    private static final int MAX_SUBSCRIBERS = 12; // Arbitrary sanity check value.
+
+    /**
+     * Corfu Runtime.
+     */
+    @Getter
+    private final CorfuRuntime runtime;
+
+    /**
+     * Map of StreamingSubscriptionThreads.
+     * <p>
+     * This is the map of all streaming subscriptions.
+     */
+    @Getter
+    private final Map<StreamListener, SubscriptionThread> subscriptions = new HashMap<>();
+
+    public StreamManager(@Nonnull CorfuRuntime runtime) {
+        this.runtime = runtime;
+    }
+
+    /**
+     * Subscribe to updates.
+     *
+     * @param streamListener   Client listener.
+     * @param namespace        Namespace of interest.
+     * @param tablesOfInterest Only updates from these tables will be returned.
+     * @param startAddress     Address to start the notifications from.
+     */
+    public synchronized <K extends Message, V extends Message, M extends Message>
```

Comment:
It would be great if you fix all yellows in the code, some of them could indicate about fragility in the code some bout bugs, thanks

![image](https://user-images.githubusercontent.com/314292/82608681-f908cc80-9b6f-11ea-964f-62c34e8ec172.png)

Reply:
looks like you forgot to add K,V,T into TableSchema signature

Reply:
done. thanks